src/decode.rs

```diff
@@ -542,6 +543,19 @@ pub struct Dav1dFrameContext_bd_fn {
     pub backup_ipred_edge: backup_ipred_edge_fn,
     pub read_coef_blocks: read_coef_blocks_fn,
 }
+
+impl Dav1dFrameContext_bd_fn {
```

Comment:
I think this type is still not yet deduplicated as it contains some bitdepth-dependent types.  So we need to be careful in adding code to it.  We'll probably want to define this on all the `bd_fn` types, for example.  But that probably could come later.  Just something to keep in mind.

Reply:
Good point. I might attempt to deduplicate this next, then, to avoid potential issues.

Reply:
To deduplicate this one, I think we'll have to do #25 with generics.  If you find a way to do it more easily, that'd be awesome.  Right now the function pointers all have `pixel`, etc. args, and `pixel` is a bitdepth-dependent type and so there may be multiple of these types existing at once.